### ISSUE TITLE
UnknownModelAttribute false-positives on migration files referencing since-dropped columns

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1251]
## Summary

`UnknownModelAttribute` checks mass-assignment calls against the schema aggregator's **final** table state (all migrations replayed in order, including later `dropColumn`/rename operations). Migration files themselves run in the past, against the schema as it stood *at that point in history* — a column can legitimately exist when a migration uses it and be dropped by a later migration. The rule has no notion of "point in time," so it flags migrations that were correct when written.

## Repro (coolify, real app)

```php
// database/migrations/2023_10_12_132431_add_standalone_redis_to_environment_variables_table.php
Schema::table('environment_variables', function (Blueprint $table) {
    $table->foreignId('standalone_redis_id')->nullable();   // ADD
});
```

```php
// database/migrations/2024_10_16_120026_move_redis_password_to_envs.php
EnvironmentVariable::create([
    'standalone_redis_id' => $redis->id,   // flagged: UnknownModelAttribute
    'key' => 'REDIS_PASSWORD',
    'value' => $redis_password,
]);
```

```php
// database/migrations/2024_12_16_134437_add_resourceable_columns_to_environment_variables_table.php
Schema::table('environment_variables', function (Blueprint $table) {
    $table->dropColumn([..., 'standalone_redis_id', ...]);   // DROP, two months later
});
```

`standalone_redis_id` existed for the ~2 months between the first and third migration. The 2024-10 migration is correct code for the schema at the time it ran. Because `MigrationSchemaBuilder` sorts all migration files by filename and replays them into one `$tables` array (`SchemaTable::dropColumn()` just does `unset($this->columns[$name])`), the column is gone from the *final* aggregate state — and every file, including migrations from before the drop, gets checked against that final state.

This is a normal, common Laravel pattern (add FK columns → later migrate to polymorphic/rename → drop the old columns), so any app with that kind of migration history will hit false positives on old data-migration files.

## Proposed fix

Exclude files under `database/migrations/**` (or wherever the configured migration path is) from `UnknownModelAttribute` reporting. Migrations are inherently point-in-time code; checking them against the current/final schema is the wrong model. This is cheap and consistent with how the rule already self-silences when a model's column schema is unknown entirely — same idea, just scoped to path instead of schema availability.

The harder alternative (resolving schema per-migration at its own timestamp instead of one final aggregate) would fix this more precisely but is a much bigger change to `SchemaAggregator`/`MigrationSchemaBuilder`, and not necessary to eliminate the false positive.

## Where

- `src/Handlers/Rules/UnknownModelAttributeHandler.php`
- `src/Handlers/Eloquent/Schema/MigrationSchemaBuilder.php` (chronological replay into one final `SchemaAggregator::$tables`)
- `src/Handlers/Eloquent/Schema/SchemaTable::dropColumn()`
